base/src/commonMain/kotlin/text/NumberFormatter.kt

```diff
@@ -0,0 +1,310 @@
+/*
+ Copyright 2020 Splendo Consulting B.V. The Netherlands
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+ */
+
+package com.splendo.kaluga.base.text
+
+import com.splendo.kaluga.base.utils.Locale
+import com.splendo.kaluga.base.utils.Locale.Companion.defaultLocale
+import kotlin.math.max
+
+/**
+ * Rounding mode applied when rounding aa given [Number] while formatting
+ */
+enum class RoundingMode {
+    /**
+     * Round towards positive infinity.
+     */
+    Ceiling,
+
+    /**
+     * Round towards negative infinity.
+     */
+    Floor,
+
+    /**
+     * Round towards zero.
+     */
+    Down,
+
+    /**
+     * Round away from zero.
+     */
+    Up,
+
+    /**
+     * Round towards the nearest integer, or towards an even number if equidistant.
+     */
+    HalfEven,
+
+    /**
+     * Round towards the nearest integer, or towards zero if equidistant.
+     */
+    HalfDown,
+
+    /**
+     * Round towards the nearest integer, or away from zero if equidistant.
+     */
+    HalfUp
+}
+
+@ExperimentalUnsignedTypes
+/**
+ * Style used for formatting a [Number] to and from a [String]
+ */
+sealed class NumberFormatStyle(open val roundingMode: RoundingMode) {
+
+    /**
+     * Formats a number to an integer representation
+     * @param minInteger The minimum number of digits to show. Defaults to `0U`.
+     * @param maxInteger The maximum number of digits to show. Defaults to [Int.MAX_VALUE].
+     * @param roundingMode The [RoundingMode] to be applied. Defaults to [RoundingMode.HalfEven].
+     */
+    data class Integer(
+        val minInteger: UInt = 0U,
+        val maxInteger: UInt = Int.MAX_VALUE.toUInt(),
```

Comment:
Integer (if it's 32 bits) has 10 digits max, so having max digits as (2^31 - 1) looks strange to me

Reply:
Updated to 309. As discussed with @avdyushin over slack, We need to consider rendering the largest possible Double as an Integer, which is 1.7976931348623157e+308, thus 309 digits for the integer, and 325 for the decimal (the decimal number itself contains 16 digits, so multiplying by 10^309 yields 325 digits)